### ISSUE TITLE
Can your model pronounce the English word?

### ISSUE THREAD

**Issue by ductho9799** [NoahDrisort/ViTacotron2#1]

**Comment by v-nhandt21** [comment]
> _No description provided._

I have included English IPA for training and handle with cross-language. However, the problem is we do not have a dataset that includes English in it. My initial proposal is to convert both Vietnamese and English into IPA representation, then train tacotron with IPA. vs (this approach not try to transcript en -> vi)

The result performs well with Vietnamese and I don't have a chance to test it on mix-code english.

If you have a quality cross-language dataset, I am really interested to support you :))

**Comment by ductho9799** [comment]
> > _No description provided._
> 
> I have included English IPA for training and handle with cross-language. However, the problem is we do not have a dataset that includes English in it. My initial proposal is to convert both Vietnamese and English into IPA representation, then train tacotron with IPA. vs (this approach not try to transcript en -> vi)
> 
> The result performs well with Vietnamese and I don't have a chance to test it on mix-code english.
> 
> If you have a quality cross-language dataset, I am really interested to support you :))

Thank you for your sharing. 
Do you have the TTS VLSP 2020 dataset? 
The VLSP  2020 dataset includes English words. You can experiment with your ViPhoneme Modules and this data.
I tried training the VietTTS dataset using Viphoneme to convert text to IPA. And then, the model can pronounce several English words such as: "fast", "Liverpool",... But It can not pronounce the specified words such as: "Real Madrid", "Vinhome", "Vinmec", "Ocean", "Park",...
Are you sure that with a dataset that included English words and trains it with phoneme transcripts, the model can pronounce the English basically?

**Comment by v-nhandt21** [comment]
> > > _No description provided._
> > 
> > 
> > I have included English IPA for training and handle with cross-language. However, the problem is we do not have a dataset that includes English in it. My initial proposal is to convert both Vietnamese and English into IPA representation, then train tacotron with IPA. vs (this approach not try to transcript en -> vi)
> > The result performs well with Vietnamese and I don't have a chance to test it on mix-code english.
> > If you have a quality cross-language dataset, I am really interested to support you :))
> 
> Thank you for your sharing.
> Do you have the TTS VLSP 2020 dataset?
> The VLSP 2020 dataset includes English words. You can experiment with your ViPhoneme Modules and this data.
> I tried training the VietTTS dataset using Viphoneme to convert text to IPA. And then, the model can pronounce several English words such as: "fast", "Liverpool",... But It can not pronounce the specified words such as: "Real Madrid", "Vinhome", "Vinmec", "Ocean", "Park",...
> Are you sure that with a dataset that included English words and trains it with phoneme transcripts, the model can pronounce the English basically?

Yes, I used to try with VLSP 2020 but this public dataset is small and not quality. Most of English word of this dataset is proper noun.

I am not sure IPA will work well for cross-language, I use eng-to-ipa  from pypi to handle english. 

To convert Vietnamese I base on wiki: https://en.wikipedia.org/wiki/Vietnamese_phonology

With my current knowledge, I think we much get dive into phonetic or G2P approach, because there is a gap between IPA of vi and en (vietnamese have tone), this much to have experimented.

As I know in this approach of Google, they try to "using a phonemic input representation to encourage sharing of model capacity across languages", I think this is interesting for us to follow: https://google.github.io/tacotron/publications/multilingual/index.html 

If you are working on an applied project, I think to solve your case as (Vinhome, Vinmec), the quick way is to transcript them to Vietnamese, I really want to know in the industry how they solve this problem, hope for sharing.

**Comment by v-nhandt21** [comment]
> > > _No description provided._
> > 
> > 
> > I have included English IPA for training and handle with cross-language. However, the problem is we do not have a dataset that includes English in it. My initial proposal is to convert both Vietnamese and English into IPA representation, then train tacotron with IPA. vs (this approach not try to transcript en -> vi)
> > The result performs well with Vietnamese and I don't have a chance to test it on mix-code english.
> > If you have a quality cross-language dataset, I am really interested to support you :))
> 
> Thank you for your sharing.
> Do you have the TTS VLSP 2020 dataset?
> The VLSP 2020 dataset includes English words. You can experiment with your ViPhoneme Modules and this data.
> I tried training the VietTTS dataset using Viphoneme to convert text to IPA. And then, the model can pronounce several English words such as: "fast", "Liverpool",... But It can not pronounce the specified words such as: "Real Madrid", "Vinhome", "Vinmec", "Ocean", "Park",...
> Are you sure that with a dataset that included English words and trains it with phoneme transcripts, the model can pronounce the English basically?

And sorry, one more thing that I need to notice that, the viphoneme version on PyPI have some bug for English, I am in the config process, and still not upload new version there. If you use viphoneme and get any error, contact me for more :heart:
